processing/src/main/java/io/druid/segment/DimensionIndexer.java

```diff
@@ -0,0 +1,323 @@
+   * On an in-memory segment (IncrementaIndex), we can determine min/max values by looking at the stream of
+   * row values seen in calls to processSingleRowValToIndexKey().
+   *
+   * However, on a disk-backed segment (QueryableIndex), the numeric dimensions do not currently have any
+   * supporting index structures that can be used to efficiently determine min/max values.
+   *
+   * When numeric dimension support is added, the segment format should be changed to store min/max values, to
+   * avoid performing a full-column scan to determine these values for numeric dims.
+   *
+   * @return min value
+   */
+  public ActualType getMinValue();
+
+
+  /**
+   * Get the maximum dimension value seen by this indexer.
+   *
+   * @return max value
+   */
+  public ActualType getMaxValue();
+
+
+  /**
+   * Get the cardinality of this dimension's values.
+   *
+   * @return value cardinality
+   */
+  public int getCardinality();
+
+
+  /**
+   * Return an object used to read rows from a StorageAdapter's Cursor.
+   *
+   * e.g. String -> DimensionSelector
+   *      Long   -> LongColumnSelector
+   *      Float  -> FloatColumnSelector
+   *
+   * See StringDimensionIndexer.makeColumnValueSelector() for a reference implementation.
+   *
+   * @param spec Specifies the output name of a dimension and any extraction functions to be applied.
+   * @param currEntry Provides access to the current TimeAndDims object in the Cursor
+   * @param desc Descriptor object for this dimension within an IncrementalIndex
+   * @return A new object that reads rows from currEntry
+   */
+  public Object makeColumnValueSelector(
+      DimensionSpec spec,
+      IncrementalIndexStorageAdapter.EntryHolder currEntry,
+      IncrementalIndex.DimensionDesc desc
+  );
+
+
+  /**
+   * Compares the row values for this DimensionIndexer's dimension from a TimeAndDims key.
+   *
+   * The dimension values within a TimeAndDims key always use the "unsorted" ordering for encoded values.
+   *
+   * The row values are passed to this function as an Object, the implementer should cast them to the type
+   * appropriate for this dimension.
+   *
+   * For example, a dictionary encoded String implementation would cast the Objects as int[] arrays.
+   *
+   * When comparing, if the two arrays have different lengths, the shorter array should be ordered first.
+   *
+   * Otherwise, the implementer of this function should iterate through the unsorted encoded values, converting
+   * them to their actual type (e.g., performing a dictionary lookup for a dict-encoded String dimension),
+   * and comparing the actual values until a difference is found.
+   *
+   * Refer to StringDimensionIndexer.compareTimeAndDimsKey() for a reference implementation.
+   *
+   * @param lhs dimension value array from a TimeAndDims key
+   * @param rhs dimension value array from a TimeAndDims key
+   * @return comparison of the two arrays
+   */
+  public int compareTimeAndDimsKey(Object lhs, Object rhs);
```

Comment:
Also this isn't really comparing a TimeAndDimsKey, it's comparing an EncodedArray